### PR TITLE
Adding stylelint documentation

### PR DESCRIPTION
**Pull request by NeOMakinG**
https://github.com/PrestaShop/PrestaShop/pull/16763 needs to be merged first

### CODE REVIEW

File: src/content/1.7/development/coding-standards/_index.md [hunk 1]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
 HTML, CSS (Sass), Twig and Smarty files MUST follow the [Mark Otto's coding standards](http://codeguide.co/).
 Mark is the creator of the [Bootstrap framework](https://getbootstrap.com/).
 
+[Stylelint](https://stylelint.io/) has been configured for the PrestaShop project to help developers to comply with these conventions.
+
+The prestashop specific configuration file can be found [here](https://github.com/PrestaShop/stylelint-config).
```
Comment (PierreRambaud) [inline src/content/1.7/development/coding-standards/_index.md]:
```suggestion
The PrestaShop configuration can be found [here](https://github.com/PrestaShop/stylelint-config).
```

File: src/content/1.7/development/coding-standards/_index.md [hunk 2]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
...
+
+Same as if you want to [compile assets][compile-assets], you need NodeJS and NPM to run it using the following command :
```
Comment (PierreRambaud) [inline src/content/1.7/development/coding-standards/_index.md]:
```suggestion
Same as if you want to [compile assets][compile-assets], you need NodeJS and NPM to run it using the following command:
```

File: src/content/1.7/development/coding-standards/_index.md [hunk 3]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
...
+
+```bash
+npm run scss-lint
+```
+
+If you want to use the stylelint's autofix functionality, you can use the following command :
```
Comment (PierreRambaud) [inline src/content/1.7/development/coding-standards/_index.md]:
```suggestion
If you want to use the autofix functionality, run this command:
```

**Comment by NeOMakinG** [comment]
By the way, wait the stylelint implementation before merging this please

File: src/content/1.7/development/coding-standards/_index.md [hunk 4]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
...
+```bash
+npm run scss-lint
+```
+
+If you want to use the autofix functionality, run this command:
+
+```bash
+npm run scss-fix
+```
+
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md]: To help developers to comply with these conventions, [Stylelint](https://stylelint.io/) has been configured in the PrestaShop project. You can find the configuration file [here](https://github.com/PrestaShop/stylelint-config).
    
    {{% notice note %}}
    Same as if you want to [compile assets]({{< ref "/1.7/development/compile-assets/_index.md" >}}), you need NodeJS and NPM to run Stylelint.
    {{% /notice  %}}
    
    You can run the linter like this:
    
    ```bash
    npm run scss-lint
    ```
    
    You can fix errors using this command:
    
    ```bash
    npm run scss-fix
    ```

File: src/content/1.7/development/coding-standards/_index.md [hunk 5]
```diff
@@ -124,4 +140,6 @@ All PrestaShop files MUST start with the PrestaShop license block:
  * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
  * International Registered Trademark & Property of PrestaShop SA
  */
-``` 
+```
+
+[compile-assets]: {{< ref "/1.7/development/compile-assets/_index.md" >}}
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md]: Note that you can't use footnotes links inside notices, so if you accept my previous suggestion you will have to delete this.

```suggestion
```

File: src/content/1.7/development/coding-standards/_index.md [hunk 6]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
 HTML, CSS (Sass), Twig and Smarty files MUST follow the [Mark Otto's coding standards](http://codeguide.co/).
 Mark is the creator of the [Bootstrap framework](https://getbootstrap.com/).
 
+To help developers to comply with these conventions, [Stylelint](https://stylelint.io/) has been configured in the PrestaShop project. You can find the configuration file [here](https://github.com/PrestaShop/stylelint-config).
+
+Same as if you want to [compile assets]({{< ref "/1.7/development/compile-assets.md" >}}), you need NodeJS and NPM to run Stylelint.
+
+You can run the linter like this:
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md]: Now that I think of it, we must think about _when_ this has become available.

```suggestion
Starting on {{< minver v="1.7.8" >}}, you can run the linter like this:
```

File: src/content/1.7/development/coding-standards/_index.md [hunk 7]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
 HTML, CSS (Sass), Twig and Smarty files MUST follow the [Mark Otto's coding standards](http://codeguide.co/).
 Mark is the creator of the [Bootstrap framework](https://getbootstrap.com/).
 
+To help developers to comply with these conventions, [Stylelint](https://stylelint.io/) has been configured in the PrestaShop project. You can find the configuration file [here](https://github.com/PrestaShop/stylelint-config).
```
Comment (Progi1984) [inline src/content/1.7/development/coding-standards/_index.md]:
```suggestion
To help developers to comply with these conventions, [Stylelint](https://stylelint.io/), a stylesheet linter, has been configured in the PrestaShop project. You can find the configuration file [here](https://github.com/PrestaShop/stylelint-config).
```

File: src/content/1.7/development/coding-standards/_index.md [hunk 8]
```diff
@@ -70,6 +70,22 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
...
+
+Same as if you want to [compile assets]({{< ref "/1.7/development/compile-assets.md" >}}), you need NodeJS and NPM to run Stylelint.
+
+Starting on {{< minver v="1.7.8" >}}, you can run the linter like this:
+
+```bash
+npm run scss-lint
+```
+
+You can fix errors using this command:
```
Comment (Progi1984) [inline src/content/1.7/development/coding-standards/_index.md]: That fix all errors or just these which are fixable ?

Comment (NeOMakinG) [reply src/content/1.7/development/coding-standards/_index.md]:
```suggestion
You can fix auto-fixable errors using this command:
```

What do you think?

Comment (Progi1984) [reply src/content/1.7/development/coding-standards/_index.md]: :ok_hand:

**Comment by mickaelandrieu** [comment]
> PrestaShop/PrestaShop#16763 needs to be merged first

It is merged, can someone merge this one ? It's free :)

**Comment by matks** [comment]
Thank you @NeOMakinG

**Pull request merged**